lib/krane/kubernetes_resource/stateful_set.rb

```diff
@@ -22,26 +22,16 @@ def status
     def deploy_succeeded?
       success = observed_generation == current_generation
 
-      @pods.each do |pod|
-        success &= pod.definition["metadata"]["labels"]["controller-revision-hash"] == status_data['updateRevision']
-      end
```

Comment:
This is the check I've removed because it's not needed as far as I can tell. This previously passed in some of our deploys due to https://github.com/kubernetes/kubernetes/issues/106055 not reliably setting `.status.currentRevision` properly.

I don't believe it should be necessary to look up the pods. The StatefulSet's status should be enough.